File: src/python/sort_builtin.py
import os
import sys


#####################
# sort script start #
#####################


def do_sorting(input_list):
    return sorted(input_list)


###################
# sort script end #
###################


if __name__ == "__main__":
    # read input file
    inputFilePath = os.getenv("INPUT_PATH")
    with open(inputFilePath, "r") as inputFileObject:
        inputFileData = inputFileObject.readlines()

    # clean input data
    cleanedInputData = []
    for element in inputFileData:
        cleanedInputData.append(element.strip())

    # do sorting
    sortedData = do_sorting(cleanedInputData)

    # write output file
    outputFilePath = os.getenv("OUTPUT_PATH")
    with open(outputFilePath, "w") as outputFileObject:
        for element in sortedData:
            outputFileObject.write(element + "\n")

File: src/rust/sort_builtin.rs
use std::env;
use std::fs;
use std::fs::File;
use std::io::Write;

//////////////////////////
// business logic start //
//////////////////////////

fn do_sorting(input_list: Vec<&str>) -> Vec<&str> {
    let mut output_list = input_list;
    output_list.sort();
    return output_list;
}

////////////////////////
// business logic end //
////////////////////////

// 👇🏽 copy pasted helpers

fn main() -> Result<(), Box<dyn std::error::Error + 'static>> {
    // setup
    let input_file_path = env::var("INPUT_PATH").unwrap();
    let output_file_path = env::var("OUTPUT_PATH").unwrap();

    // get input data
    let input_data_string = fs::read_to_string(input_file_path)?;
    let input_data_vec = input_data_string.lines().collect();

    // do sorting
    let sorted_data = do_sorting(input_data_vec);

    // write output data
    // join on \n, and add an additional trailing \n
    let output_data_string = sorted_data.join("\n") + "\n";
    let output_data_bytes = output_data_string.as_bytes();
    let mut output_file = File::create(output_file_path)?;
    output_file.write_all(output_data_bytes)?;

    // teardown
    Ok(())
}
